### PR TITLE
Cloud UI: show dashboards in a table

### PR DESCRIPTION
**Pull request by ericpgreen2**
This PR switches the display of a project's dashboards from a list to a table. This is a virtualized table that can be searched and, in the future, filtered and sorted. Additionally, in the display, we now include the time when a dashboard was last refreshed.

Note: the table is currently left-aligned. The PR to add project tabs (https://github.com/rilldata/rill/pull/3240) will center the table.

### Summary by CodeRabbit

- New Feature: Added a comprehensive table view for dashboards, enhancing data organization and accessibility.
- New Feature: Implemented search functionality in the dashboard table, allowing users to quickly find specific dashboards.
- New Feature: Introduced a 'time ago' function, providing users with a human-readable timestamp for dashboard updates.
- Improvement: Enhanced error handling with specific messages when dashboards fail to load, improving user experience during troubleshooting.
- Improvement: Updated the dashboard page layout with better padding and spacing, enhancing the visual appearance.
- Improvement: Added a message and link to documentation when no dashboards are found, guiding users on how to deploy a dashboard.
- Style: Updated various components with improved styling for better user interface consistency.

### CODE REVIEW

File: web-admin/src/features/dashboards/listing/utils.ts
Comment (djbarnwal) [inline web-admin/src/features/dashboards/listing/utils.ts:7]: Nit: Why don't we use our existing methods and framework around time for this? (Luxon and Durations)

File: web-common/src/components/search/Search.svelte
Comment (djbarnwal) [inline web-common/src/components/search/Search.svelte:36]: These changes are affecting search components at other places

**Review by djbarnwal — CHANGES_REQUESTED** [review]
The search bar for filter pills, comparison selector, visible measure selector etc. looks misaligned and has an outline - 

image

image

File: web-common/src/components/search/Search.svelte
Comment (ericpgreen2) [reply web-common/src/components/search/Search.svelte:36]: Ah, good catch, just fixed the padding in 3 other Search instances. Janet created this [new Search component](https://www.figma.com/file/nqqazRo1ckU9ooC9ym9weI/Rill-Design-System?type=design&node-id=78-203&mode=design&t=sGyyC9hgxQtE13gy-0) that this code moves towards.

File: web-admin/src/features/dashboards/listing/utils.ts
Comment (ericpgreen2) [reply web-admin/src/features/dashboards/listing/utils.ts]: Good idea to use `luxon` – just switched. But is this code redundant of something already in our codebase? I looked a little bit but didn't see a method that could be dropped in here.

File: web-common/src/components/search/Search.svelte
Comment (ericpgreen2) [reply web-common/src/components/search/Search.svelte:37]: Note that the blue outline is from the focus state, and these search instances are currently set to focus-on-mount.

**Review by djbarnwal — APPROVED** [review]
Looks good.

One thing which still bugs me is the blue border on the search bar. It's too in the eye if I haven't clicked on the search bar itself. Maybe we can have a prop for if we want to enable the blue border or not.
image

Anyway that's a separate issue. I would leave it up to you. Approving so that you can merge at your convenience.

**Comment by ericpgreen2** [comment]
> One thing which still bugs me is the blue border on the search bar. It's too in the eye if I haven't clicked on the search bar itself. Maybe we can have a prop for if we want to enable the blue border or not.

Totally agree. We do have that prop in `autofocus`. Just turned off `autofocus` and aesthetics look better.

**Pull request merged**
